explorer/websockethandlers.go

```diff
@@ -59,55 +60,107 @@ func (exp *explorerUI) RootWebsocket(w http.ResponseWriter, r *http.Request) {
 
 				// handle received message according to event ID
 				var webData WebSocketMessage
+				//  If the request sent is past the limit exit the loop
+				if len(msg.Message) > requestLimit {
+					log.Debug("Request size over limit")
+					webData.Message = "Request too large"
+					continue
+				}
+
 				switch msg.EventId {
 				case "decodetx":
-					webData.EventId = msg.EventId + "Resp"
-					if len(msg.Message) > requestLimit {
-						log.Debug("Request size over limit")
-						webData.Message = "Request too large"
-						break
-					}
 					log.Debugf("Received decodetx signal for hex: %.40s...", msg.Message)
 					tx, err := exp.blockData.DecodeRawTransaction(msg.Message)
 					if err == nil {
 						message, err := json.MarshalIndent(tx, "", "    ")
 						if err != nil {
 							log.Warn("Invalid JSON message: ", err)
-							webData.Message = fmt.Sprintf("Error: Could not encode JSON message")
+							webData.Message = "Error: Could not encode JSON message"
 							break
 						}
 						webData.Message = string(message)
 					} else {
 						log.Debugf("Could not decode raw tx")
 						webData.Message = fmt.Sprintf("Error: %v", err)
 					}
+
 				case "sendtx":
-					webData.EventId = msg.EventId + "Resp"
-					if len(msg.Message) > requestLimit {
-						log.Debugf("Request size over limit")
-						webData.Message = "Request too large"
-						break
-					}
 					log.Debugf("Received sendtx signal for hex: %.40s...", msg.Message)
 					txid, err := exp.blockData.SendRawTransaction(msg.Message)
 					if err != nil {
 						webData.Message = fmt.Sprintf("Error: %v", err)
 					} else {
 						webData.Message = fmt.Sprintf("Transaction sent: %s", txid)
 					}
-				case "getmempooltxs":
-					webData.EventId = msg.EventId + "Resp"
 
+				case "getmempooltxs":
```

Comment:
Please change the below Lock/Unlock to RLock/RUnlock while you are here.

Reply:
ok. Noted